src/psyclone/tests/psyir/frontend/fortran_test.py

```diff
@@ -86,46 +88,68 @@ def test_fortran_psyir_from_source():
 def test_fortran_psyir_from_expression(fortran_reader):
     ''' Test that the psyir_from_expression method generates the
     expected PSyIR. '''
-    psyir = fortran_reader.psyir_from_expression("3.0")
+    table = SymbolTable()
+    psyir = fortran_reader.psyir_from_expression("3.0", table)
     assert isinstance(psyir, Literal)
     assert psyir.value == "3.0"
-    psyir = fortran_reader.psyir_from_expression("-3.0 + 1.0")
+    psyir = fortran_reader.psyir_from_expression("-3.0 + 1.0", table)
     assert isinstance(psyir, BinaryOperation)
     assert psyir.operator == BinaryOperation.Operator.ADD
     assert isinstance(psyir.children[0], UnaryOperation)
     assert psyir.children[0].operator == UnaryOperation.Operator.MINUS
     assert isinstance(psyir.children[0].children[0], Literal)
     assert psyir.children[0].children[0].value == "3.0"
-    psyir = fortran_reader.psyir_from_expression("ABS(-3.0)")
+    psyir = fortran_reader.psyir_from_expression("ABS(-3.0)", table)
     assert isinstance(psyir, UnaryOperation)
     assert psyir.operator == UnaryOperation.Operator.ABS
     assert isinstance(psyir.children[0], UnaryOperation)
     assert psyir.children[0].operator == UnaryOperation.Operator.MINUS
     assert isinstance(psyir.children[0].children[0], Literal)
-    with pytest.raises(NotImplementedError) as err:
-        fortran_reader.psyir_from_expression("3.0 + a")
-    assert "Expression must contain only literals: '3.0 + a'" in str(err.value)
-    with pytest.raises(NotImplementedError) as err:
-        fortran_reader.psyir_from_expression("3.0 + sin(a)")
-    assert ("Expression must contain only literals: '3.0 + sin(a)'" in
+    # With kind specified by a kind parameter. Add a ContainerSymbol with a
+    # wildcard import so there's some way that 'r_def' can be brought into
+    # scope.
+    csym = table.new_symbol("kind_params_mod", symbol_type=ContainerSymbol)
+    csym.wildcard_import = True
+    fortran_reader.psyir_from_expression("3.0_r_def", table)
+    assert "r_def" in table
+    psyir = fortran_reader.psyir_from_expression("3.0_r_def", table)
+    assert isinstance(psyir, Literal)
+    assert isinstance(psyir.datatype, ScalarType)
+    assert isinstance(psyir.datatype.precision, DataSymbol)
+    assert psyir.datatype.precision is table.lookup("r_def")
+    # Symbol not found in supplied table
+    with pytest.raises(SymbolError) as err:
```

Comment:
See the comment about this error in `fortran.py` about the meaning of "unresolved". From the expression used here it seems to me that the error is raised after the process of creating symbols and when a symbol cannot be found in a supplied symbol table. Is this correct?

Reply:
Yes, that's correct - symbols are created before the executable statements are processed.